src/core/datastore/cachestore.js

```diff
@@ -154,6 +154,9 @@ export class CacheStore extends NetworkStore {
    * @return  {Promise}                                                         Promise
    */
   pendingSyncCount(query, options) {
+    if (query) {
+      return this.syncManager.getSyncItemCountByEntityQuery(this.collection, query);
```

Comment:
Some time ago, I raised the question of querying the sync queue and the way it works. Because essentially when you are using methods like `pendingSyncCount`, `clearSync` and other sync queue related methods, they all accept a query. And that query doesn't regard the actual sync items, but the entities they are about. So it is applied to the collection (of entities) and whatever entities match, a query to the sync items collection (aka the sync queue) is made, for items which are about those entity ids. There is a problem with this, that deleted entities would not get matched, as they are not in their collection. So the sync items for those entities will not get "used". That's a bug. We don't yet have a solution for this. In order to not change functionality, I've kept those public methods working as they used to, but introduced this distinction. Where it says "entity query", it's a query which works on entities, because I think if you just call it query, one might assume it works on sync items.

The method `getSyncItemCount` doesn't take a query, because I didn't want to mix the functionality which should get fixed, with the functionality that works as expected - it just gets all the sync items (including ones for deleted entities).

Reply:
That distinction makes sense. "Pushing a subset of changes" is something we've been asked about, which is why the methods with queries exist. 
I see the problem with deletes. As you said, separate topic :)